### PR TITLE
Command-line interface for exporting MPS languages to LionWeb JSON

### PR DESCRIPTION
**Pull request by enikao**

### CODE REVIEW

File: solutions/io.lionweb.mps.cmdline/readme.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+usage: lionweb-export-language <project-dir> [<language-file>] <output-file>
+ -help                 print this message
+ -l,--language <arg>   convert language with key <args>
+ -m,--macro <arg>      set macro (a.k.a. path variable)
+ -s,--scope <arg>      set export scope: listed, indirect, fineGrainedClosure
+```
+
+* _project-dir:_ The root directory of the MPS project to load.
+
+* _language-file:_ Optional; if provided, it must be a json file with two properties:
```
Comment (dslmeinte) [inline solutions/io.lionweb.mps.cmdline/readme.md]: This language file and the `language` and `scope` CLI arguments play the same role: are they mutually exclusive in some way?

File: solutions/io.lionweb.mps.cmdline/readme.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  * `scope` takes the same values as `-s` parameter. If both are present, the parameter prevails.
+  * `languages` contains a list of language keys to export.
```
Comment (dslmeinte) [inline solutions/io.lionweb.mps.cmdline/readme.md:25]: "[...]a list of (fully-qualified) names of languages", I guess? Or is the LionWeb-key attribute used here?

**Review by dslmeinte — APPROVED** [review]

**Comment by dslmeinte** [comment]
I think it'd be good to have the `test-project` project and the other project at the same level in the repository, and not as a sub dir. of `solutions`. We can make multiple dir.s inside `mps/` at the time of repo restructuring.

File: solutions/io.lionweb.mps.cmdline/readme.md (hunk 1 above)
Comment (enikao) [reply solutions/io.lionweb.mps.cmdline/readme.md]: The languages are additive, for scope the command-line wins (is mentioned there). Should this be mentioned more explicitly?

File: solutions/io.lionweb.mps.cmdline/readme.md (hunk 2 above)
Comment (enikao) [reply solutions/io.lionweb.mps.cmdline/readme.md:25]: Nope, LionWeb-key

Comment (dslmeinte) [reply solutions/io.lionweb.mps.cmdline/readme.md:25]: Would be good to document.

File: solutions/io.lionweb.mps.cmdline/readme.md (hunk 1 above)
Comment (dslmeinte) [reply solutions/io.lionweb.mps.cmdline/readme.md]: Would be good to document.

**Comment by enikao** [comment]
> I think it'd be good to have the `test-project` project and the other project at the same level in the repository, and not as a sub dir. of `solutions`. We can make multiple dir.s inside `mps/` at the time of repo restructuring.

I don't get this:

* `test-project` is its own subdirectory, not inside any `solutions` folder
* This is the lionweb-mps repo, I hope we won't restructure this?

**Comment by dslmeinte** [comment]
> > I think it'd be good to have the `test-project` project and the other project at the same level in the repository, and not as a sub dir. of `solutions`. We can make multiple dir.s inside `mps/` at the time of repo restructuring.
> 
> I don't get this:
> 
>     * `test-project` is its own subdirectory, not inside any `solutions` folder
> 
>     * This is the lionweb-mps repo, I hope we won't restructure this?

Indeed: sorry, context mixup on my side – just forget; my bad.

**Review by dslmeinte — APPROVED** [review]

**Pull request merged**